File: src/app/src/main/java/com/example/screenservicetvapp/ObjectExtensions.java
package com.example.screenservicetvapp;

import android.content.Intent;
import android.os.Parcelable;

public class ObjectExtensions {
    public static boolean isNullOrEmpty(String input) {
        return input == null || input.equals("");
    }

    public static <T extends Parcelable> T[] getParcelableArrayExtra(Intent intent, String key, Class<T> type) {
        Parcelable[] array = intent.getParcelableArrayExtra(key);

        if (array != null && type.isInstance(array[0])) {
            T[] result = (T[]) java.lang.reflect.Array.newInstance(type, array.length);

            for (int i = 0; i < array.length; i++) {
                //noinspection unchecked
                result[i] = (T) array[i];
            }

            return result;
        } else {
            // Handle the case where the array is null or the type doesn't match
            return null;
        }
    }
}

File: src/app/src/main/java/com/onscreensync/tvapp/utils/ObjectExtensions.kt
package com.onscreensync.tvapp.utils

import android.content.Intent
import android.os.Build
import android.os.Parcelable

/**
 * Extension functions and utility methods for object manipulations.
 */

object ObjectExtensions {

    @JvmStatic
    fun isNullOrEmpty(str: String?): Boolean {
        return str == null || str.trim().isEmpty() || str == "null"
    }

    @JvmStatic
    fun convertToInt(str: String?, defaultValue: Int = 0): Int {
        if (isNullOrEmpty(str)) return defaultValue
        return try {
            str!!.toInt()
        } catch (e: NumberFormatException) {
            defaultValue
        }
    }

    @JvmStatic
    fun <T : Parcelable> getParcelableArrayExtra(intent: Intent, name: String, clazz: Class<T>): Array<T>? {
        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            intent.getParcelableArrayExtra(name, clazz)
        } else {
            @Suppress("DEPRECATION")
            val parcelableArray = intent.getParcelableArrayExtra(name)
            if (parcelableArray != null) {
                java.lang.reflect.Array.newInstance(clazz, parcelableArray.size) as? Array<T>
            } else {
                null
            }
        }
    }
}

// Extension functions for convenience
fun String?.isNullOrEmpty(): Boolean = ObjectExtensions.isNullOrEmpty(this)
fun String?.convertToInt(defaultValue: Int = 0): Int = ObjectExtensions.convertToInt(this, defaultValue)
